hershensh.cpp

```diff
@@ -0,0 +1,30 @@
+#include <iostream>
+
+#include "evaluate.h"
+#include "printer.h"
+#include "reader.h"
+
+using namespace hershensh;
+
+const std::string defaultPrompt = "hershensh$";
+
+bool isEOF() {
```

Comment:
I prefer if we have the calling function above the called function. Can we do that in this project?
Also, do you have a c++ style guide that we can follow? Most projects I've seen do
```
bool isEOF()
{
<>
}
```
and not
```
bool isEOF() {
<>
}

Reply:
That doesn't really make sense to me - if you read the file from top to bottom, I'd expect to see the function declaration before it's usage.
Furthermore, some programming languages like python require you to define the function before using it. So I think this is a good pattern to use.

I wouldn't say there is a typical bracing pattern. See for example in the [Catch2 framework](https://github.com/catchorg/Catch2/blob/devel/src/catch2/catch_session.cpp).

I don't have a particular style guide I like, but if you find something interesting, do share.

Reply:
No one selects a file and reads it from top to bottom. You start from the function that you're interested in and decide if you want to dive into the implementation details of the functions that it calls. If you do it the way you're doing - you will have to read going up and not going down. See the "reading code from top to bottom..." section in the chapter about functions in clean code.